### DISCUSSION TITLE
Cannot access some folders from file explorer

### DISCUSSION THREAD

**Discussion by tasso85** [microsoft/WSL#6949] [Q&A]
Hi everyone, I hope I can have some help with this issue.

I have installed WSL2 on my development machine, and I have installed Ubuntu-20.04 from the store.

I can access the files in the linux filesystem by opening a file explored at `\\wsl$\Ubuntu-20.04`, however, I cannot access some of the folders, for example if I try to open `/root` I get the following error:

![immagine](https://user-images.githubusercontent.com/655560/118472215-e1bfd900-b708-11eb-86b4-47103a84f092.png)

While the text is in Italian, it says that it cannot open `\\wsl$\Ubuntu-20.04\root` because I do not have the necessary permissions, however on this machine I am a member of the Administrator group.

Is there a way to access all of the WSL filesystem from Windows Explorer?

**Comment by Baalkikhaal** [comment]
Hi @tasso85,
I am able to access the root of my `Ubuntu-18.04` distribution instance by entering the following command via address bar of Windows explorer.

    \\wsl.localhost\Ubuntu-18.04

I did a search on [WSL documentation](https://docs.microsoft.com/en-us/windows/wsl/) with the keyword `wsl.localhost`. It turns out that since [build 21364](https://docs.microsoft.com/en-us/windows/wsl/release-notes#build-21364) the recommended way to access Linux file system from Windows has changed from `\\wsl` to `\\wsl.localhost`.

Quoting from [release notes of 21364](https://blogs.windows.com/windows-insider/2021/04/07/announcing-windows-10-insider-preview-build-21354/)

> The default path to access Linux files from Windows using the Windows Subsystem for Linux has been changed from `\\wsl\\` to `\\wsl.localhost\\` to improve performance and reliability. You will still be able to access Linux files using `\\wsl$\\`.

Please ensure your machine is rolling above 21364 release by executing `winver` at the Windows command shell or `winver.exe` at bash shell.

**Reply by tasso85** [reply]
How would I check which build I have? I just tried `\\wsl.localhost` on the file explorer, but it gave an error saying that it does not exist, while `\\wsl$` works (notice the final `$`).

Also notice that, for example, I can correctly access `\\wsl$\Ubuntu-20.04\home\mtassinari` (my home folder on WSL), but not `\\wsl$\Ubuntu-20.04\root` (that is, the home folder of the `root` user).

**Reply by Baalkikhaal** [reply]
> Also notice that, for example, I can correctly access `\\wsl$\Ubuntu-20.04\home\mtassinari` (my home folder on WSL), but not `\\wsl$\Ubuntu-20.04\root` (that is, the home folder of the `root` user).

Sorry I misunderstood root to mean the root of the filesystem `/` 😜 . Yes you are right, even I am not able to access the root folder of the WSL; even running

    > explorer \\wsl.localhost\Ubuntu-18.04\root

with an **elevated Command prompt**. I too get the same message

![image](https://user-images.githubusercontent.com/8896597/119861906-87f3b600-bf35-11eb-8b4a-ebfe6123f99d.png)

This is an interesting situation where the host administrator is not able to access some files 😕 .

> How would I check which build I have?

You can run

    > winver

in command prompt.

**Reply by Baalkikhaal** [reply]
> > Also notice that, for example, I can correctly access `\\wsl$\Ubuntu-20.04\home\mtassinari` (my home folder on WSL), but not `\\wsl$\Ubuntu-20.04\root` (that is, the home folder of the `root` user).
> 
> Sorry I misunderstood root to mean the root of the filesystem `/` 😜 . Yes you are right, even I am not able to access the root folder of the WSL; even running

From the [File Permissions](https://docs.microsoft.com/en-us/windows/wsl/file-permissions#accessing-linux-files-from-windows-using-wsl) 

> Accessing Linux files via `\\wsl$` will use the default user of your WSL distribution. Therefore any Windows app accessing Linux files will have the same permissions as the default user.

So to access the `/root` folder of Ubuntu-20.04, the default user needs to be set to `root`. 

Firstly, we need ensure that the distribution instance is **not** running. 

```bash
D:\WSL_Backups>wsl --list --running
Windows Subsystem for Linux Distributions:
Ubuntu-18.04 (Default)
Ubuntu-copy
```
In case we have opened the Ubuntu filesystem in explorer, the instance is actually running. So we need to exit the filesystem also.

Now, from [WSL commands](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#change-the-default-user-for-a-distribution), execute the following in a windows ~~ubuntu~~ shell

    D:\>ubuntu2004 config --default-user root

**Reply by tasso85** [reply]
Setting the default user to root fixed it, thanks!
